x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,366 @@
+          defaultMessage:
+            'Connectors with `is_native: true` are only allowed in Elastic-managed environments',
+        }),
+        message: i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.message', {
+          defaultMessage:
+            '"Native Connectors" are only intended for Elastic-managed environments like Elastic Cloud and ' +
+            'Elastic Serverless. Any connectors with `is_native: true` must be converted to connector clients or deleted ' +
+            'before this upgrade can proceed.',
+        }),
+        documentationUrl:
+          'https://elastic.co/guide/en/enterprise-search/current/upgrading-to-9-x.html',
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.listConnectors', {
+              defaultMessage: 'Enumerate all connector records',
+            }),
+            i18n.translate('xpack.enterpriseSearch.deprecations.notManaged.convertPretenders', {
+              defaultMessage: '"Convert to Client" any that have `is_native: true`.',
+            }),
+          ],
+          api: {
+            method: 'POST',
+            path: '/internal/enterprise_search/deprecations/convert_connectors_to_client',
+            body: {
+              ids: nativeConnectors.map((it) => it.id),
+            },
+          },
+        },
+      });
+    } else {
+      const nativeServiceTypes = [
+        'azure_blob_storage',
+        'box',
+        'confluence',
+        'dropbox',
+        'github',
+        'gmail',
+        'google_cloud_storage',
+        'google_drive',
+        'jira',
+        'mssql',
+        'mongodb',
+        'mysql',
+        'network_drive',
+        'notion',
+        'onedrive',
+        'oracle',
+        'outlook',
+        'postgresql',
+        's3',
+        'salesforce',
+        'servicenow',
+        'sharepoint_online',
+        'sharepoint_server',
+        'slack',
+        'microsoft_teams',
+        'zoom',
+      ];
+      const nativeTypesStr = '[' + nativeServiceTypes.join(', ') + ']';
+      const fauxNativeConnectors = nativeConnectors.filter(
+        (hit) => !nativeServiceTypes.includes(hit.service_type!)
+      );
+
+      if (fauxNativeConnectors.length > 0) {
+        // There are some illegal service_types in their native connectors
+        deprecations.push({
+          level: 'critical',
+          deprecationType: 'feature',
+          title: i18n.translate('xpack.enterpriseSearch.deprecations.fauxNativeConnector.title', {
+            defaultMessage: 'Connectors with `is_native: true` must be of supported service types',
+          }),
+          message: i18n.translate(
+            'xpack.enterpriseSearch.deprecations.fauxNativeConnector.message',
+            {
+              values: { serviceTypes: nativeTypesStr },
+              defaultMessage:
+                'Only specific service types are supported for Elastic-managed connectors. ' +
```

Comment:
```suggestion
                'Elastic-managed connectors don't support all service types. ' +
```

Reply:
my 2 cents, "don't support all service types" sounds a bit awkward
How about 
_Elastic-managed connectors are currently available for a select range of service types_....

Reply:
agree about my first suggestion, but I think we can get to the point quicker, so edited my edit :)